### PR TITLE
[processor/transform] added span link ottlcontext

### PR DESCRIPTION
**Pull request by fujin**
**Description:** <Describe what has changed.>

Copied the ottlspanevent stuff to make it so we can filter ottlspanlinks.

**Link to tracking Issue:** <Issue number if applicable>
#26660 

**Testing:** <Describe what testing was performed and which tests were added.>
Not yet

**Documentation:** <Describe the documentation added.>
Not yet

### CODE REVIEW

**Comment by fujin** [comment]
@TylerHelmuth is this what you meant?

File: pkg/ottl/contexts/ottlspanlink/span_links.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+	case "attributes":
+		mapKey := path[0].Keys
+		if mapKey == nil {
+			return accessSpanLinkAttributes(), nil
+		}
+		return accessSpanLinkAttributesKey(mapKey), nil
+	// TODO(fujin): Would it be valid to enable modifying the span id, is this the trace.link.span_id? It doesn't seem like there is something in 
+	// https://github.com/open-telemetry/opentelemetry-collector/blob/main/pdata/ptrace/generated_spanlink.go that allows it to be set.
+	// case "span_id":
+	// 	return accessSpanLinkSpanID(), nil
```
Comment (TylerHelmuth) [inline pkg/ottl/contexts/ottlspanlink/span_links.go:128]: Yes, we should allow modifying it.  These functions should allow setting spanlink traceID and spanID: https://github.com/open-telemetry/opentelemetry-collector/blob/53615832e65416a89667611481b8765ab244021c/pdata/ptrace/generated_spanlink.go#L52-L72

**Review by TylerHelmuth — COMMENTED** [review]
@fujin this is the right implementation, but we should separate the ottlcontext addition and its consumption by the transform processor into 2 PRs.  Lets have this PR be the addition of the new context, and then once it is merged we can update various components to use the new context.

**Comment by fujin** [comment]
Cool, will do!

**Comment by TylerHelmuth** [comment]
@fujin you still interested in this PR?

**Comment by fujin** [comment]
yeah I am! sorry- on call shift last week! sometime this week?

**Comment by TylerHelmuth** [comment]
@fujin great! When you're ready for another view take it out of draft.

**Comment by fujin** [comment]
I'm still trying to move this along. I added the trace_id/span_id handling, still need to do tests.

**Pull request closed without merging**